File: Assignment_11/Program5/program5.java
////////////////////////////////////////////////////////////////////////////////
// 
// File: program5.java
// Description: Used to accept a nuber from user and display below pattern.
// Input: 8
// Output: 2 4 6 8 10 12 14 16
// Author: Parimal Kelkar
// Date: 26/05/2025
// 
////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class program5
{
    public static void main(String args[])
    {
        Scanner sObj = new Scanner(System.in);

        int iValue = 0;

        System.out.println("Enter number: ");
        iValue = sObj.nextInt();

        Pattern pObj = new Pattern();
        pObj.Display(iValue);
        
    }
}

class Pattern
{
    ////////////////////////////////////////////////////////////////////////////
    //
    // Method: Display()
    // Description: Accepts a number and displays the pattern 2 4 6 8 10 12 14 16 if input number is 8.
    // Input: int
    // Output: void
    // Author: Parimal Kelkar
    // Date: 26/05/2025
    //
    ////////////////////////////////////////////////////////////////////////////
    void Display(int iNo)
    {
        int iCnt = 0;
        char ch = '\0';

        // Updator
        if(iNo < 0)
        {
            iNo = -iNo;
        }

        for(iCnt = 1; iCnt <= iNo; iCnt++)
        {
            System.out.print((iCnt * 2) + "\t");
        }
    }
}

File: Assignment_41/program5/program5.c
////////////////////////////////////////////////////////////////////////////////
//
//  File name: Program5.c
//  Description: Used to print below pattern recursively.
//  Input: 6
//  Output: a b c d e f
//  Author: Parimal Kelkar
//  Date: 22/07/2025
//
////////////////////////////////////////////////////////////////////////////////

#include<stdio.h>

////////////////////////////////////////////////////////////////////////////////
//
//  Name: Display()
//  Description: Displays a pattern recursively.
//  Input: int
//  Output: void
//  Author: Parimal Kelkar
//  Date: 22/07/2025
//
////////////////////////////////////////////////////////////////////////////////

void Display(int iNo)
{
    static char ch = 'a';

    if(iNo > 0)
    {
        printf("%c\t", ch++);
        Display(--iNo);
    }
}

int main()
{
    int iValue = 0;

    printf("Enter number: ");
    scanf("%d", &iValue);

    Display(iValue);

    return 0;
}